### ISSUE TITLE
identify SoloLTR

### ISSUE THREAD

**Issue by Bioyu721** [oushujun/EDTA#279]
Dear Shujun,
    Can you give us a script to identify soloLTR based on the annotation file of EDTA,the script of LTRretriver to identify solo_LTR can hardly suit the outcome for this software

**Comment by Bioyu721** [comment]
Dear Shujun,
The similar script like intact_finder.pl and solo_intact_raio.pl of LTR_retriever are also need to calculate,could you provide these based on the EDTA annotation outcome. I tried to use the middle file 'rawLTR.scn' to run the LTR_retriever,but I find the outcome of annotation is  very different from the EDTA.

**Comment by Bioyu721** [comment]
Dear Shujun,
Another question is how do we define the high Removal efficiency of solo/Intact ratio,from an essay of nature communication,some defined the more the families which S/I ratio greater than 3,  the higher the removal efficiency of the genome. I don't know whether I'm wrong

**Comment by oushujun** [comment]
Hello @Bioyu721,

Thank you for your interest in finding solo LTR and calculate Solo-Intact ratio. I have an active project that works with this type of data and we should be able to release it soon. I will update EDTA by then.

Regarding the S/I ratio, I don't think it's studied enough to draw a line for fast or slow removal. You may reference other studies or set up your own rule based on your observations. Let me know what you think.

Best,
Shujun

**Comment by Bioyu721** [comment]
> Hello @Bioyu721,
> 
> Thank you for your interest in finding solo LTR and calculate Solo-Intact ratio. I have an active project that works with this type of data and we should be able to release it soon. I will update EDTA by then.
> 
> Regarding the S/I ratio, I don't think it's studied enough to draw a line for fast or slow removal. You may reference other studies or set up your own rule based on your observations. Let me know what you think.
> 
> Best, Shujun
       I'm excited to receive your reply. My idea of S/I ratio is from Cell.2021 DOI: 10.1016/j.cell.2021.12.006, the literature of pine genome and azaleas genome from  Nature Communication.2020. DOI: 10.1038/s41467-020-18771-4.I used tools like cd-hit to classify all the Intact LTR into families or extraxct the 'Name:TE' from file of ".EDTA.intact.gff3" . Then observed the  fragment LTR in 'TEanno.gff3' with the same Name,so I can know the num of solo LTR in each family. I'm sorry to express not clearly bacause it's too complicated. 
       I do some tasks to use TEsorter to classify the Intact LTR into clade such as CRM,Galadriel and so on. I do some tasks to  compare the num of different genomes of LTR clade. I wanna know the change of LTR whether it is related to the recombination rate of solo LTR.  Totally ,S/I ratio in many literature directly or indirectly uncovers the reason why the genome is too large or small or LTR removal. But lack the same standard of identify the solo LTR. From the view of scientific rigor, I wanna use the script to identify the solo LTR from EDTA annotation. Thanks,Shujun
      Yours sincerely,
      Zhipeng

**Comment by Bioyu721** [comment]
Dear @oushujun 
I have commented in your reply,but I forgot @ you,I'm sorry about that and I'm a rookie. How to identify these Solo LTR,I have no idea and I' waiting to your project and software updating.
I'm excited to receive your reply. My idea of S/I ratio is from Cell.2021 DOI: 10.1016/j.cell.2021.12.006, the literature of pine genome and azaleas genome from Nature Communication.2020. DOI: 10.1038/s41467-020-18771-4.I used tools like cd-hit to classify all the Intact LTR into families or extraxct the 'Name:TE' from file of ".EDTA.intact.gff3" . Then observed the fragment LTR in 'TEanno.gff3' with the same Name,so I can know the num of solo LTR in each family. I'm sorry to express not clearly bacause it's too complicated.
I do some tasks to use TEsorter to classify the Intact LTR into clade such as CRM,Galadriel and so on. I do some tasks to compare the num of different genomes of LTR clade. I wanna know the change of LTR whether it is related to the recombination rate of solo LTR. Totally ,S/I ratio in many literature directly or indirectly uncovers the reason why the genome is too large or small or LTR removal. But lack the same standard of identify the solo LTR. From the view of scientific rigor, I wanna use the script to identify the solo LTR from EDTA annotation. Thanks,Shujun
Another quetion about Intact LTR, these LTR I think are the LTR from the result file of '.EDTA.intact.gff3' or 'pass.list' and these can calculated the insertion time. But your script of intact_coarse.pl is about the relative intact LTR. I'm not sure which kind of Intact LTR should be viewed as intact LTR to calculate.

Yours sincerely,
Zhipeng

**Comment by oushujun** [comment]
Hi Zhipeng,

No worries I will receive all comments because I am the owner of this repo.

Currently the script to find solo LTR is not adapted to EDTA results. You may want to modify the script to make it fit. 

The script to find relatively intact LTR is "intact_coarse.pl", and as indicated by the name, it's coarse. It aims to include those partially complete elements (ie., presence of left and right LTR regions) because they have not gone through the illegitimate recombination process to become a solo LTR. You may use the ones from the pass.list, but you will have higher levels of missing data.

For solo/intact ratio, you can make your own rule, as far as the comparison is fair. For fair comparisons, you need to make sure the assembly quality of those being compared is similar. For example, the Illumina-based spruce genome may not be comparable to the PacBio-based maize genome, because TE assemblies in the spruce genome are much worse and most intact LTR elements are likely not assembled. You may use the LAI value to check for the overall TE assembly quality.

Shujun

**Comment by oushujun** [comment]
Using 'raw_LTR.scn' will get more intact LTR than the `pass.list` generated by LTR_retriever and more than the list generated by `intact_coarse.pl`. It may be a good solution. Please try it out and let me know. The solo LTR method I am working with is just a modification of the original script but adapted to EDTA outputs, which is part of a bigger project that I hope I will release it in a month or two.

**Comment by Bioyu721** [comment]
Hi @oushujun 
Thanks to your advice,I have tried this solution, I used 'rawLTR.scn' by running LTR_retriever and I find there are more fragment LTRs in annotation file so that we can identify more coarse intact LTR, and the total proportion in genome changed not obviously.  In its 'pass.list' file, I find the same counts of intact LTR compared with 'pass.list' in EDTA. In my opinion, no matter which annotation file I use, I don't think it's wrong to research S/I ratio.
Yours sincerely,
Zhipeng

**Comment by oushujun** [comment]
Yes, the number of coarse intact LTR will be `intact_coarse.pl` > `rawLTR.scn` > `pass.list`. EDTA uses LTR_retriever to identify LTRs, so the number of pass.list from the two outputs are the same. Looking forward to your new findings!

Best,
Shujun

**Comment by Bioyu721** [comment]
Dear Shujun,
I use the result of coarse_intact.pl to identify the relative intact_LTR, finally,I use the Intact LTR of 'pass.list' with the relative Intact LTR as total intact LTRs of the whole genome. Do you think whether I'm wrong.
Why I do this is beacause the solo_intact_ratio.pl's outcome has too many INF, how to deal with these INF with fam. But in 'pass.list' file,it has the intact LTR by domain search.
Best zhipeng

**Comment by oushujun** [comment]
Hi Zhipeng,

You are doing it right. And you should expect lots of INFs because you are calculating solo/intact ratio. If you don't have any intact elements in this family, you get an INF. To alleviate this issue, you may instead calculate the intact/solo ratio, but then you may get INFs when no solo is found for some families.

Shujun

**Comment by Orz-CQ** [comment]
Hi Shujun,

I am interested to find the solo LTR in the EDTA's output.
Here I am thinking about the two different ways.

1st, since the EDTA provided the `genome.fa.mod.EDTA.intact.gff3` and `genome.fa.mod.EDTA.TEanno.gff3`.  Could I just count the non-overlapping part of the  `genome.fa.mod.EDTA.TEanno.gff3` and further regarded them as solo LTR?

2nd, maybe I need to try the `solo_finder.pl` from` LTR_retriever.` Through the perl script, if the `RepeatMasker.out` can be the output from EDTA, `genome.fa.out.EDTA.TEanno.out` ?  
I am also not sure which file generated by `EDTA` could be used for the `-info lib.LTR.info`

Best,
Lan

**Comment by oushujun** [comment]
The 2nd approach should get you more valid ones. I believe I have some
notes on how to use the script or explanations for inputs. Please check out
the script’s content or the github issues/wiki. Let me know if you still
need help.

Shujun

**Comment by oushujun** [comment]
I am revisiting this thread, and seems like the instruction was not clear. I am sorry. 

Here are more details on calculating solo:intact LTR ratios:

`perl ~/bin/LTR_retriever/bin/find_LTR.pl -lib TElib.fa > TElib.fa.info`
`perl ~/bin/LTR_retriever/bin/solo_finder.pl -i genome.RepeatMasker.out -info TElib.fa.info > genome.RepeatMasker.solo.list`
`perl ~/bin/LTR_retriever/bin/intact_finder_coarse.pl genome.RepeatMasker.out > genome.RepeatMasker.intact.list`
`perl ~/bin/LTR_retriever/bin/solo_intact_ratio.pl genome.RepeatMasker.solo.list genome.RepeatMasker.intact.list > genome.RepeatMasker.solo_intact.ratio`

If you wish to use EDTA results, you can use the EDTA.TElib.fa as the input library, and EDTA.anno/*EDTA.RM.out file as the RepeatMasker out file. 

I also write a little wrapper for all these steps:
`sh ~/bin/LTR_retriever/bin/solo_intact_ratio_wrap.sh TElib.fa genome.RepeatMasker.out`

I hope it helps!

Shujun

**Comment by oushujun** [comment]
Codes were updated in https://github.com/oushujun/LTR_retriever/commit/266d2fbb0a1c14b4d7ff4e67098688a41f929899

**Comment by Orz-CQ** [comment]
Appreciated for your help!

Lan
